sdk-setup/src/mb2

```diff
@@ -1799,7 +1799,25 @@ sign_rpms() {
     if [[ -x $sdksign ]]; then
         $sdksign ${rpms}
     else
-        rpmsign --addsign ${rpms}
+        if [ ! -z "$OPT_PACKAGE_SIGN_USER" ]; then
+            #check that key is exported
+            gpg2 --list-keys $OPT_PACKAGE_SIGN_USER >> /dev/null
+            if [ $? -ne 0 ]; then
+                notice "No GPG key known as $OPT_PACKAGE_SIGN_USER. First import the secret key!"
+                exit $?
+            fi
+
+            echo "$OPT_PACKAGE_SIGN_PASSPHRASE" | gpg2 -q --batch --status-fd 1 --sign --local-user "$OPT_PACKAGE_SIGN_USER" --passphrase-fd 0 &> /dev/null
+            if [ $? -ne 0 ]; then
+                notice "GPG key has passphrase protection. Set correct passphrase with \"--package-signing-passphrase\" option."
+                exit $?
+            fi
+
+            echo -e "${OPT_PACKAGE_SIGN_PASSPHRASE}\n${OPT_PACKAGE_SIGN_PASSPHRASE}\n" | rpm --addsign ${rpms} --define "_gpg_name $OPT_PACKAGE_SIGN_USER" \
+                --define "__gpg_sign_cmd %{__gpg} gpg --no-verbose --batch --passphrase-fd 0 -u '%{_gpg_name}' -sbo %{__signature_filename} %{__plaintext_filename}"
+        else
+            rpm --addsign ${rpms}
+        fi
```

Comment:
I forgot to ask: Is it *really* needed to pass passphrase 2 times? (can be done with `3<<<"$passphrase"$'\n'"$passphrase"` if so).

Reply:
`--passphrase-fd 3" 3<<<"$passphrase"`

Dosn't work with rpmsign --define and other ways to use --passphrase-fd 3 I couldn't run on the current version of rpm and gpg

Two times password need because rpmsign ask password twice per file